main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -420,18 +422,43 @@ object Lowering extends Phase[Root, Root] {
     case Expression.NewChannel(exp, tpe, eff, loc) =>
       val e = visitExp(exp)
       val t = visitType(tpe)
-      Expression.NewChannel(e, t, eff, loc)
+
+      val enumSym = Symbol.mkEnumSym("Channel.ChannelImpl")
+      val chanType = Type.mkEnum(enumSym, List(t))
+      val defTpe = Type.mkImpureArrow(Type.Int32, chanType)
+      val sym = Symbol.mkDefnSym("Channel.newWithCapacity")
```

Comment:
Unless I am missing something, I only see `Channel.newWithCapacity_duplicate`? But maybe you are using a different Channel.flix locally?